src/aspire/utils/rotation.py

```diff
@@ -391,3 +391,31 @@ def angle_dist(r1, r2, dtype=None):
             theta = max(min(theta, 1), -1)  # Clamp theta in [-1,1]
             dist = np.arccos(theta, dtype=dtype)
         return dist
+
+    @staticmethod
+    def mean_angular_distance(rots_1, rots_2, dtype=None):
+        """
+        Find the mean angular distance between two sets of rotation matrices.
+
+        :param rots_1: An nx3x3 array of rotation matrices.
```

Comment:
Shouldn't this be taking `Rotation` objects as input?

Reply:
This is mainly a diagnostic tool that is used on `src.rotations` which is implemented in `ImageSource` as `self._rotations.matrices`, ie. as a `numpy.ndarray`.

Will add support for `Rotation` objects as well.

Reply:
Support for `Rotation` objects added as part of having `angle_dist` take care of stacks.

Reply:
Ok cool.